ui/src/buckets/selectors/index.ts

```diff
@@ -5,28 +5,36 @@ export const SYSTEM = 'system'
 
 export const isSystemBucket = (type: string): boolean => type === SYSTEM
 
-export const getSortedBucketNames = (buckets: Bucket[]) =>
-  buckets
-    .sort((a, b) => {
-      const firstBucket = `${a.name}`.toLowerCase()
-      const secondBucket = `${b.name}`.toLowerCase()
-      if (firstBucket === secondBucket) {
-        return 0
-      }
-      if (isSystemBucket(a.type)) {
-        // ensures that the default system types are the last buckets
-        return 1
-      }
-      if (isSystemBucket(b.type)) {
-        // ensures that the default system types are the last buckets
-        return -1
-      }
-      if (firstBucket < secondBucket) {
-        return -1
-      }
-      if (firstBucket > secondBucket) {
-        return 1
-      }
-      return 0
-    })
-    .map(bucket => bucket.name)
+const sortFunc = (a: Bucket, b: Bucket) => {
+  const firstBucket = `${a.name}`.toLowerCase()
+  const secondBucket = `${b.name}`.toLowerCase()
+  if (firstBucket === secondBucket) {
+    return 0
+  }
+  if (firstBucket < secondBucket) {
+    return -1
+  }
+  if (firstBucket > secondBucket) {
+    return 1
+  }
+  return 0
+}
+
+export const getSortedBucketNames = (buckets: Bucket[]) => {
+  const systemBuckets = []
+  const otherBuckets = []
+  buckets.forEach(bucket => {
+    // separate system buckets from the rest
+    if (isSystemBucket(bucket.type)) {
+      systemBuckets.push(bucket)
+    } else {
+      otherBuckets.push(bucket)
+    }
+  })
+  // alphabetize system buckets
+  systemBuckets.sort(sortFunc)
```

Comment:
All of `sortFunc` could be replaced with 
```
_.sortBy((bucket) => bucket.name.toLocaleLowerCase())
```

Reply:
I very weirdly can't get it to work using the sortBy function